tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -3097,7 +3096,29 @@ std::optional<Value> convertReduceOpCommon(
                        /*scale32=*/true);
   }
 
+  // If keep dims, no reshaping of the output is required
+  if (keep_dims) {
+    return val;
+  }
+
   // Squeeze out the reduced axes.
+  const auto squeeze_axes = [](llvm::ArrayRef<int64_t> in, llvm::ArrayRef<int64_t> axes) {
+    llvm::SmallVector<int64_t> sorted_axes{axes};
+    std::sort(sorted_axes.begin(), sorted_axes.end());
+    auto current_axis = sorted_axes.begin();
+
+    llvm::SmallVector<int64_t> out;
+    out.reserve(in.size() - axes.size());
+    for (const auto& [i, dim] : llvm::enumerate(in)) {
+      if (current_axis == sorted_axes.end() || i != *current_axis)
+        out.push_back(dim);
+      else
+        current_axis++;
+    }
+    return out;
+  };
```

Comment:
The logic here covers the case where we are reducing across multiple axis, but there are no tests for such case. I think it would be worth adding one.

Reply:
Very interesting, there is indeed! How did this test pass before the squeezing logic was added?

Reply:
Previously it was just relying on the specified output type of the tf op, which is static here so naturally becomes the correct type after lowering. The new squeeze logic computes the same shape here but from the input types, so the test didn’t need updating